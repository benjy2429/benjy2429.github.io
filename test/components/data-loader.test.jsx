/**
 * @jest-environment jsdom
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import dataLoader from '../../src/components/data-loader';

const Component = () => <div>wrapped component</div>;
const dataFile = { foo: 'foo' };

describe('Data loader component', () => {
  it('renders wrapped component with data', () => {
    const WrappedComponent = dataLoader(Component, dataFile);
    const output = shallow(<WrappedComponent bar="bar" />);
    expect(toJson(output)).toMatchSnapshot();
  });
});
