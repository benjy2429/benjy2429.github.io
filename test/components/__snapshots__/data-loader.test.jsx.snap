// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data loader component renders wrapped component with data 1`] = `
<Component
  bar="bar"
  foo="foo"
/>
`;
